jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.cpp

```diff
@@ -733,25 +734,66 @@ ecma_date_make_day (ecma_number_t year, /**< year value */
     return ecma_number_make_nan ();
   }
 
+  /* 2., 3., 4. */
   ecma_number_t y = ecma_number_trunc (year);
   ecma_number_t m = ecma_number_trunc (month);
   ecma_number_t dt = ecma_number_trunc (date);
+  /* 5. */
   ecma_number_t ym = y + (ecma_number_t) floor (m / 12);
+  /* 6. */
   ecma_number_t mn = (ecma_number_t) fmod (m, 12);
   mn = (mn < 0) ? 12 + mn : mn;
+
+  /* 7. */
   ecma_number_t time = ecma_date_time_from_year (ym);
+  ecma_number_t step = floor ((ecma_number_t) (365) / 2);
+  ecma_number_t delta = step;
+
+  /**
+   * The algorithm below searches the following date: ym-mn-1
+   * To find this time it starts from the beggining of the year (ym)
+   * then increase it by ECMA_DATE_MS_PER_DAY until it reaches the
+   * right date.
+   */
+  if (ecma_date_year_from_time (time) == ym)
```

Comment:
The `Date` needs 52 significant bit to represent the date as the standard describes it. Probably it would be better to use `int64_t` especially on those target where the `ecma_numer_t` is single precision. I think this refactoring should be in different PR with other optimizations as well, but I'm not sure how much performance gain we would get through optimizing the `Date` object.